runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -133,7 +139,7 @@
      */
     @Deprecated // TODO: Add replacement method that conforms to style
     @SuppressWarnings("checkstyle:abbreviation") // Due to deprecation
-    public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[] args,
+    public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, String streamName, Class<T> type, Object[] args,
```

Comment:
Its cleaner to pass down the stream name and then derive the stream id. Passing down redundant pieces of information that can be derived is error-prone.

Reply:
Sounds good